### ISSUE TITLE
Feature Request: Rule for Flagging Missing Required Parameters in Function Calls

### ISSUE THREAD

**Issue by gtupak** [astral-sh/ruff#10234]
Would it be possible to add a rule in Ruff to detect and flag missing required parameters when calling a function? 

This feature would enhance error detection, especially in complex projects, by ensuring that all necessary arguments are provided during function calls. Implementing such a rule could prevent runtime errors related to missing arguments and improve code robustness. 

Here's an example:

```python
def test(required_param: str):
    print(required_param)

test() # This should raise an error
```

**Comment by MichaReiser** [comment]
Agree, that would be useful. Although I'm not sure if that would be covered by a type checker instead #3893

**Comment by JohannesBuchner** [comment]
@gtupak you can try [pystrict3](https://github.com/JohannesBuchner/pystrict3/)

**Comment by gtupak** [comment]
@JohannesBuchner I'm already using pylance but it would be great to have this in ruff.

I'll keep following the progress on this in https://github.com/astral-sh/ruff/issues/3893
